index.js

```diff
@@ -37,16 +37,14 @@ function build () {
       }
     }
 
-    emitted.set(code, false)
     codes[code] = buildWarnOpts
-
     return codes[code]
   }
 
   function emit (code, a, b, c) {
+    if (emitted[code] === true) return
```

Comment:
The main performance gain is the `return` line moved above the check of `codes` exist.
Since the function here is simple enough, one more check may means double of time.